src/enclave/main.cpp

```diff
@@ -59,6 +61,13 @@ extern "C"
     }
 #endif
 
+    // Host and enclave versions must match. Otherwise the node may crash much
+    // later (e.g. unhandled ring buffer message on either end)
+    if (std::string(host_version) != ccf::ccf_version)
+    {
+      return false;
```

Comment:
+1 for a custom return code, this is too difficult to debug otherwise.

Reply:
Done